### PR TITLE
Mod Options Fixes for Below Zero

### PR DESCRIPTION
**Pull request by MrPurple6411**
### Changes made in this pull request

  - Fixed Mod Options not finding Mods Tab
  - Fixed Null Key Get in ModOptionsHeadingsToggle at StatesConfig.get
  - Updated some dependencies.

### CODE REVIEW

File: SMLHelper/Patchers/OptionsPanelPatcher.cs [hunk 1]
```diff
@@ -132,7 +142,17 @@ private class StatesConfig
 
                     public HeadingState this[string name]
                     {
-                        get => states.TryGetValue(name, out HeadingState state)? state: HeadingState.Expanded;
+                        get
+                        {
+                            if(string.IsNullOrEmpty(name))
```
Comment (zorgesho) [inline SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: Why this check is needed here?

File: SMLHelper/Options/SliderModOption.cs [hunk 2]
```diff
@@ -6,6 +6,9 @@
     using UnityEngine;
     using UnityEngine.UI;
     using UnityEngine.Events;
+#if BELOWZERO
```
Comment (zorgesho) [inline SMLHelper/Options/SliderModOption.cs:9]: Can it be something like this?
```cs
#if SUBNAUTICA
    using Text = UnityEngine.UI.Text;
#endif
#if BELOWZERO
    using Text = TMPro.TextMeshProUGUI;
#endif
```

Comment (PrimeSonic) [reply SMLHelper/Options/SliderModOption.cs:9]: This should also work. Personally I don't have a preference for either style.

File: SMLHelper/Patchers/OptionsPanelPatcher.cs (hunk 1 above)
Comment (MrPurple6411) [reply SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: somehow in BZ a null is getting passed into this getter.

File: SMLHelper/Options/SliderModOption.cs (hunk 2 above)
Comment (MrPurple6411) [reply SMLHelper/Options/SliderModOption.cs]: yup this works as well. and as I have no issue either way I put that in.

Comment (zorgesho) [reply SMLHelper/Options/SliderModOption.cs]: Less duplicated code this way

File: SMLHelper/Patchers/OptionsPanelPatcher.cs (hunk 1 above)
Comment (PrimeSonic) [reply SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=netframework-4.0

An `ArgumentNullException` can be thrown if it's passed into `Dictionary.TryGetValue`.  
This seems like a sensible check to make.

Comment (MrPurple6411) [reply SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: exactly what was happening.

Comment (zorgesho) [reply SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: Yeah, probably `text` field in new BZ Text component can be null if it is empty.

@MrPurple6411 Let it be something like this then:
`states.TryGetValue(name ?? "", out HeadingState state)? state: HeadingState.Expanded;`

Comment (MrPurple6411) [reply SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: ahhhh ok thats the issue... Text is no longer used in BZ  its now TextMeshProUGUI   that would do it. meaning i need to go find your ref to Text that is getting passed to this.

Comment (zorgesho) [reply SMLHelper/Patchers/OptionsPanelPatcher.cs:147]: It's there https://github.com/SubnauticaModding/SMLHelper/blob/f59e6ec623fd39fe3697a6e676306c0a4125d994/SMLHelper/Patchers/OptionsPanelPatcher.cs#L219

**Pull request merged**